client_test.go

```diff
@@ -1015,3 +1015,370 @@ func TestUpdateSegment(t *testing.T) {
 	assert.Equal(t, "1", segment.Rules[0].Rules[0].Conditions[0].Value)
 
 }
+
+const EnvironmentID int64 = 100
+const EnvironmentAPIKey = "environment_api_key"
+const EnvironmentJson = `{
+	"id": 100,
+	"name": "Development",
+	"api_key": "environment_api_key",
+	"description": null,
+	"project": 10,
+	"minimum_change_request_approvals": 0,
+	"allow_client_traits": true
+}`
+
+func TestGetEnvironment(t *testing.T) {
+	// Given
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		assert.Equal(t, fmt.Sprintf("/api/v1/environments/%s/", EnvironmentAPIKey), req.URL.Path)
+		assert.Equal(t, "GET", req.Method)
+		assert.Equal(t, "Api-Key "+MasterAPIKey, req.Header.Get("Authorization"))
+
+		rw.Header().Set("Content-Type", "application/json")
+		_, err := io.WriteString(rw, EnvironmentJson)
+		assert.NoError(t, err)
+	}))
+	defer server.Close()
+
+	client := flagsmithapi.NewClient(MasterAPIKey, server.URL+"/api/v1")
+
+	// When
+	environment, err := client.GetEnvironment(EnvironmentAPIKey)
+
+	// Then
+	// assert that we did not receive an error
+	assert.NoError(t, err)
+
+	// assert that the environment is as expected
+	assert.Equal(t, EnvironmentID, environment.ID)
+	assert.Equal(t, "Development", environment.Name)
+	assert.Equal(t, EnvironmentAPIKey, environment.APIKey)
+	assert.Equal(t, ProjectID, environment.Project)
+}
+
+// 400 is arbitrarily chosen to avoid collision with other ids
+const FeatureSegmentID = int64(400)
+const GetFeatureSegmentJson = `{
+	"id": 400,
+	"uuid": "7b7bbb74-00bc-4d14-aabe-3d44debe4662",
+	"segment": 300,
+	"priority": 0,
+	"environment": 100,
+	"segment_name": "is_not_set",
+	"is_feature_specific": false
+}`
+
+const CreateFeatureSegmentJson = `{
+	"id": 400,
```

Comment:
We shouldn't provide the ID for creating the FeatureSegment, right?

Reply:
That's actually the response, not the request. I will update the variable name here